### PR TITLE
add a DeepGreen colormap

### PR DESCRIPTION
**Pull request by jsxyhelu**
### This pullrequest changes
I implemented the colormap "DeepGreen"
And, I add cv::COLORMAP_DEEPGREEN as the flag of cv::applyColorMap.
I checked this feature using the following code.
```
#include <opencv2/core.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/imgcodecs.hpp>

int main(int argc, const char* argv[])
{
    cv::Mat disp(cv::Size(256, 30), CV_8UC1, cv::Scalar(0));
    for(int y = 0; y < disp.rows; y++)
    {
        for(int x = 0; x < disp.cols; x++)
        {
            disp.at<uchar>(y, x) = x;
        }
    }
    cv::Mat deepgreen;
    cv::applyColorMap(disp, deepgreen, cv::COLORMAP_DEEPGREEN);
    cv::imwrite("colorscale_deepgreen.jpg", deepgreen);
    return 0;
}
```
colorscale_deepgreen

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch

### CODE REVIEW

**Comment by jsxyhelu** [comment]
why the autobot is always blocked?

**Comment by jsxyhelu** [comment]
why the autobot is always blocked??

**Comment by catree** [comment]
@jsxyhelu You have to create a branch with a specific name.
See:
- [How to contribute to the OpenCV repository](https://github.com/opencv/opencv/wiki/How_to_contribute)
- [Branches](https://github.com/opencv/opencv/wiki/Branches)

If you have a **clean local state** on your computer, it is something like:
```
# 3.4 branch
git checkout 3.4
# create new branch from 3.4
git checkout -b feat_deep_green_colormap
```

**Have a look at these introduction tutorials**: 
- [Fork a repo](https://help.github.com/en/github/getting-started-with-github/fork-a-repo)
- [Syncing a fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork)

---

Can you add the visualisation of the colormap both in this pull request and in the documentation?
**See this for the code**: #15388

Also, do you have a reference on this colormap? Something like an equivalent in Matplotlib, gnuplot, or Matlab for instance?

If this is a custom colormap, I would advice if it is possible to use an existing colormap. See for instance the [**colormap in Matplotlib**](https://matplotlib.org/tutorials/colors/colormaps.html).

File: modules/imgproc/src/colormap.cpp [hunk 1]
```diff
@@ -758,6 +780,7 @@ namespace colormap
             colormap == COLORMAP_TWILIGHT ? (colormap::ColorMap*)(new colormap::Twilight) :
             colormap == COLORMAP_TWILIGHT_SHIFTED ? (colormap::ColorMap*)(new colormap::TwilightShifted) :
             colormap == COLORMAP_VIRIDIS ? (colormap::ColorMap*)(new colormap::Viridis) :
+            colormap == COLORMAP_DEEPGREEN ? (colormap::ColorMap*)(new colormap::DeepGreen) :
```
Comment (catree) [inline modules/imgproc/src/colormap.cpp:783]: Minor.
Can you keep it sorted alphabetically?

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 2]
```diff
@@ -4376,7 +4376,8 @@ enum ColormapTypes
     COLORMAP_CIVIDIS = 17, //!< ![cividis](pics/colormaps/colorscale_cividis.jpg)
     COLORMAP_TWILIGHT = 18, //!< ![twilight](pics/colormaps/colorscale_twilight.jpg)
     COLORMAP_TWILIGHT_SHIFTED = 19, //!< ![twilight shifted](pics/colormaps/colorscale_twilight_shifted.jpg)
-    COLORMAP_TURBO = 20 //!< ![turbo](pics/colormaps/colorscale_turbo.jpg)
+    COLORMAP_TURBO = 20, //!< ![turbo](pics/colormaps/colorscale_turbo.jpg)
+    COLORMAP_DEEPGREEN = 21
```
Comment (catree) [inline modules/imgproc/include/opencv2/imgproc.hpp:4380]: Visualisation is missing.

File: modules/imgproc/src/colormap.cpp (hunk 1 above)
Comment (jsxyhelu) [reply modules/imgproc/src/colormap.cpp]: you are right,I have changed my code.

File: modules/imgproc/include/opencv2/imgproc.hpp (hunk 2 above)
Comment (jsxyhelu) [reply modules/imgproc/include/opencv2/imgproc.hpp]: Visualisation  is added.

File: modules/imgproc/src/colormap.cpp (hunk 1 above)
Comment (jsxyhelu) [reply modules/imgproc/src/colormap.cpp]: #15388 is great!many thanks!@catree

**Pull request closed without merging**